tests/php/Dev/Tasks/TagsToShortcodeHelperTest.php

```diff
@@ -0,0 +1,429 @@
+<?php
+
+namespace SilverStripe\Assets\Tests\Dev\Tasks;
+
+use Silverstripe\Assets\Dev\TestAssetStore;
+use SilverStripe\Assets\Filesystem;
+use SilverStripe\Assets\Dev\Tasks\TagsToShortcodeHelper;
+use SilverStripe\Assets\File;
+use SilverStripe\Assets\Folder;
+use SilverStripe\Assets\Image;
+use SilverStripe\Assets\Tests\Dev\Tasks\Shortcode\HtmlObject;
+use SilverStripe\Assets\Tests\Dev\Tasks\Shortcode\NoStage;
+use SilverStripe\Assets\Tests\Dev\Tasks\Shortcode\SubHtmlObject;
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Core\Environment;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\FieldType\DBHTMLVarchar;
+use SilverStripe\Versioned\Versioned;
+
+class TagsToShortcodeHelperTest extends SapphireTest
+{
+    protected static $fixture_file = 'TagsToShortcodeHelperTest.yml';
+
+    protected static $extra_dataobjects = [
+        HtmlObject::class,
+        SubHtmlObject::class,
+        NoStage::class
+    ];
+
+    /**
+     * get the BASE_PATH for this test
+     *
+     * @return string
+     */
+    protected function getBasePath()
+    {
+        // Note that the actual filesystem base is the 'assets' subdirectory within this
+        return ASSETS_PATH . '/TagsToShortcodeHelperTest';
+    }
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->setupAssetStore();
+    }
+
+    private function setupAssetStore()
+    {
+        // Set backend root to /TagsToShortcodeHelperTest/assets
+        TestAssetStore::activate('TagsToShortcodeHelperTest/assets');
+
+        // Ensure that each file has a local record file in this new assets base
+
+        $destinations = [];
+
+        /** @var File $file */
+        foreach (File::get()->filter('ClassName', File::class) as $file) {
+            $file->setFromString($file->getFilename(), $file->getFilename(), $file->getHash());
+        }
+
+        $from = __DIR__ . '/../../ImageTest/test-image-low-quality.jpg';
+        foreach (Image::get() as $file) {
+            $file->setFromLocalFile($from, $file->getFilename(), $file->getHash());
+            $file->setFromLocalFile($from, $file->getFilename(), $file->getHash(), 'ResizedImageWzY0LDY0XQ');
+        }
+    }
+
+    public function tearDown()
+    {
+        TestAssetStore::reset();
+        Filesystem::removeFolder($this->getBasePath());
+        parent::tearDown();
+    }
+
+    public function testRewrite()
+    {
+        $tagsToShortcodeHelper = new TagsToShortcodeHelper();
+        $tagsToShortcodeHelper->run();
+
+        /** @var SiteTree $newPage */
```

Comment:
Just as a sidenote: I think this isn't great usage of fixtures - it makes it quite hard to see the before/after state. I generally try to avoid fixtures where the data of those records is relevant to understanding the tests, and default to creating them inline. No need to change here, just thought it's worth mentioning.